bad_cert_demo3.js

```diff
@@ -0,0 +1,24 @@
+var tls = require('tls');
+var https = require('https');
+
+var socket3 = tls.connect({
+    port: 1337,
+    host: 'https://example1.com',
+    rejectUnauthorized: true,
+    checkServerIdentity: function(servername, peer) {  
```

Comment:
Coverity found issue: TLS uses bad certificate verification - CWE-295, Medium Severity

**Improper certificate verification may allow an attacker to spoof a trusted entity by using a man-in-the-middle attack.**

**How to fix:** Use the default implementation either implicitly or explicitly, as the default function provided by the "tls" module is secure. When using a custom implementation, make sure that the function throws an error when the certificate does not match, and not just returns "undefined" on every input.
<details>
<summary>Click to expand data flow...</summary>
**From bad_cert_demo3.js:**
```
    5     port: 1337,
    6     host: 'https://example1.com',
    7     rejectUnauthorized: true,
1. tls: The "checkServerIdentity" property in the "tls.connect()" function uses bad cert verification.
2. remediation: Use the default implementation either implicitly or explicitly, as the default function provided by the "tls" module is secure. When using a custom implementation, make sure that the function throws an error when the certificate does not match, and not just returns "undefined" on every input.
    8     checkServerIdentity: function(servername, peer) {  
    9         const good = '77:53:28:AD:42:B1:04:F7:49:2B:C7:C7:7B:2A:84:64:EA:0B:1F:CE';
   10         const bad = 'mismatch';
```
</details><!-- Coverity 055c1a5f38b3326bf18c0ea5b9fcef43 -->